[2025-03-11T21:11:44.188+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-11T21:11:44.199+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: redfin_analytics_dag.tsk_transform_redfin_data manual__2025-03-11T21:02:19.139782+00:00 [queued]>
[2025-03-11T21:11:44.206+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: redfin_analytics_dag.tsk_transform_redfin_data manual__2025-03-11T21:02:19.139782+00:00 [queued]>
[2025-03-11T21:11:44.206+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 3
[2025-03-11T21:11:44.220+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): tsk_transform_redfin_data> on 2025-03-11 21:02:19.139782+00:00
[2025-03-11T21:11:44.226+0000] {standard_task_runner.py:72} INFO - Started process 17518 to run task
[2025-03-11T21:11:44.229+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'redfin_analytics_dag', 'tsk_transform_redfin_data', 'manual__2025-03-11T21:02:19.139782+00:00', '--job-id', '108', '--raw', '--subdir', 'DAGS_FOLDER/redfin_analytics.py', '--cfg-path', '/tmp/tmps7eqq3_0']
[2025-03-11T21:11:44.229+0000] {standard_task_runner.py:105} INFO - Job 108: Subtask tsk_transform_redfin_data
[2025-03-11T21:11:44.273+0000] {task_command.py:467} INFO - Running <TaskInstance: redfin_analytics_dag.tsk_transform_redfin_data manual__2025-03-11T21:02:19.139782+00:00 [running]> on host ip-172-31-7-71.eu-west-2.compute.internal
[2025-03-11T21:11:44.329+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='dengineer420@gmail.com' AIRFLOW_CTX_DAG_OWNER='Rahul' AIRFLOW_CTX_DAG_ID='redfin_analytics_dag' AIRFLOW_CTX_TASK_ID='tsk_transform_redfin_data' AIRFLOW_CTX_EXECUTION_DATE='2025-03-11T21:02:19.139782+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-11T21:02:19.139782+00:00'
[2025-03-11T21:11:44.330+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-03-11T21:11:44.330+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-03-11T21:11:44.331+0000] {logging_mixin.py:190} INFO - Current task name:tsk_transform_redfin_data state:running start_date:2025-03-11 21:11:44.199953+00:00
[2025-03-11T21:11:44.331+0000] {logging_mixin.py:190} INFO - Dag name:redfin_analytics_dag and current dag run status:running
[2025-03-11T21:11:44.331+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-11T21:11:44.351+0000] {credentials.py:670} INFO - Found credentials in shared credentials file: ~/.aws/credentials
[2025-03-11T21:11:46.464+0000] {logging_mixin.py:190} INFO - Number of rows:  78387
[2025-03-11T21:11:46.465+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:48.179+0000] {logging_mixin.py:190} INFO - Number of rows:  78389
[2025-03-11T21:11:48.179+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:49.972+0000] {logging_mixin.py:190} INFO - Number of rows:  78363
[2025-03-11T21:11:49.973+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:51.754+0000] {logging_mixin.py:190} INFO - Number of rows:  78201
[2025-03-11T21:11:51.754+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:53.618+0000] {logging_mixin.py:190} INFO - Number of rows:  78693
[2025-03-11T21:11:53.618+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:55.302+0000] {logging_mixin.py:190} INFO - Number of rows:  78486
[2025-03-11T21:11:55.303+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:57.196+0000] {logging_mixin.py:190} INFO - Number of rows:  78707
[2025-03-11T21:11:57.197+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:11:59.019+0000] {logging_mixin.py:190} INFO - Number of rows:  78690
[2025-03-11T21:11:59.020+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:00.766+0000] {logging_mixin.py:190} INFO - Number of rows:  78540
[2025-03-11T21:12:00.766+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:02.625+0000] {logging_mixin.py:190} INFO - Number of rows:  78598
[2025-03-11T21:12:02.626+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:04.539+0000] {logging_mixin.py:190} INFO - Number of rows:  78437
[2025-03-11T21:12:04.540+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:06.601+0000] {logging_mixin.py:190} INFO - Number of rows:  78420
[2025-03-11T21:12:06.601+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:08.656+0000] {logging_mixin.py:190} INFO - Number of rows:  78406
[2025-03-11T21:12:08.656+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:10.387+0000] {logging_mixin.py:190} INFO - Number of rows:  78494
[2025-03-11T21:12:10.387+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:12.502+0000] {logging_mixin.py:190} INFO - Number of rows:  78203
[2025-03-11T21:12:12.502+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:14.287+0000] {logging_mixin.py:190} INFO - Number of rows:  78348
[2025-03-11T21:12:14.288+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:16.066+0000] {logging_mixin.py:190} INFO - Number of rows:  78111
[2025-03-11T21:12:16.067+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:17.942+0000] {logging_mixin.py:190} INFO - Number of rows:  78130
[2025-03-11T21:12:17.942+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:19.990+0000] {logging_mixin.py:190} INFO - Number of rows:  77966
[2025-03-11T21:12:19.990+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:21.819+0000] {logging_mixin.py:190} INFO - Number of rows:  78157
[2025-03-11T21:12:21.820+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:23.642+0000] {logging_mixin.py:190} INFO - Number of rows:  78121
[2025-03-11T21:12:23.642+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:25.470+0000] {logging_mixin.py:190} INFO - Number of rows:  78119
[2025-03-11T21:12:25.470+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:27.322+0000] {logging_mixin.py:190} INFO - Number of rows:  78099
[2025-03-11T21:12:27.322+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:29.256+0000] {logging_mixin.py:190} INFO - Number of rows:  78339
[2025-03-11T21:12:29.256+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:31.033+0000] {logging_mixin.py:190} INFO - Number of rows:  78282
[2025-03-11T21:12:31.033+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:33.058+0000] {logging_mixin.py:190} INFO - Number of rows:  78265
[2025-03-11T21:12:33.059+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:35.160+0000] {logging_mixin.py:190} INFO - Number of rows:  77985
[2025-03-11T21:12:35.161+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:36.946+0000] {logging_mixin.py:190} INFO - Number of rows:  78035
[2025-03-11T21:12:36.946+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:38.786+0000] {logging_mixin.py:190} INFO - Number of rows:  78086
[2025-03-11T21:12:38.786+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:40.531+0000] {logging_mixin.py:190} INFO - Number of rows:  77905
[2025-03-11T21:12:40.532+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:42.429+0000] {logging_mixin.py:190} INFO - Number of rows:  78090
[2025-03-11T21:12:42.429+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:44.327+0000] {logging_mixin.py:190} INFO - Number of rows:  78029
[2025-03-11T21:12:44.327+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:46.250+0000] {logging_mixin.py:190} INFO - Number of rows:  78193
[2025-03-11T21:12:46.250+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:48.278+0000] {logging_mixin.py:190} INFO - Number of rows:  77752
[2025-03-11T21:12:48.278+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:50.027+0000] {logging_mixin.py:190} INFO - Number of rows:  77774
[2025-03-11T21:12:50.027+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:51.948+0000] {logging_mixin.py:190} INFO - Number of rows:  77638
[2025-03-11T21:12:51.949+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:53.906+0000] {logging_mixin.py:190} INFO - Number of rows:  77894
[2025-03-11T21:12:53.906+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:55.616+0000] {logging_mixin.py:190} INFO - Number of rows:  78018
[2025-03-11T21:12:55.616+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:57.618+0000] {logging_mixin.py:190} INFO - Number of rows:  77576
[2025-03-11T21:12:57.618+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:12:59.365+0000] {logging_mixin.py:190} INFO - Number of rows:  77483
[2025-03-11T21:12:59.365+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:01.193+0000] {logging_mixin.py:190} INFO - Number of rows:  77590
[2025-03-11T21:13:01.194+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:03.375+0000] {logging_mixin.py:190} INFO - Number of rows:  77711
[2025-03-11T21:13:03.375+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:05.209+0000] {logging_mixin.py:190} INFO - Number of rows:  77745
[2025-03-11T21:13:05.209+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:07.022+0000] {logging_mixin.py:190} INFO - Number of rows:  77575
[2025-03-11T21:13:07.023+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:08.859+0000] {logging_mixin.py:190} INFO - Number of rows:  77553
[2025-03-11T21:13:08.859+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:10.968+0000] {logging_mixin.py:190} INFO - Number of rows:  77368
[2025-03-11T21:13:10.969+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:12.781+0000] {logging_mixin.py:190} INFO - Number of rows:  76973
[2025-03-11T21:13:12.781+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:14.909+0000] {logging_mixin.py:190} INFO - Number of rows:  77285
[2025-03-11T21:13:14.909+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:16.840+0000] {logging_mixin.py:190} INFO - Number of rows:  77209
[2025-03-11T21:13:16.841+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:18.780+0000] {logging_mixin.py:190} INFO - Number of rows:  77154
[2025-03-11T21:13:18.780+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:20.677+0000] {logging_mixin.py:190} INFO - Number of rows:  77136
[2025-03-11T21:13:20.677+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:22.656+0000] {logging_mixin.py:190} INFO - Number of rows:  77072
[2025-03-11T21:13:22.656+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:24.648+0000] {logging_mixin.py:190} INFO - Number of rows:  77022
[2025-03-11T21:13:24.648+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:26.704+0000] {logging_mixin.py:190} INFO - Number of rows:  76953
[2025-03-11T21:13:26.705+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:28.488+0000] {logging_mixin.py:190} INFO - Number of rows:  76854
[2025-03-11T21:13:28.489+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:30.297+0000] {logging_mixin.py:190} INFO - Number of rows:  76596
[2025-03-11T21:13:30.297+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:32.167+0000] {logging_mixin.py:190} INFO - Number of rows:  76021
[2025-03-11T21:13:32.168+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:13:33.626+0000] {logging_mixin.py:190} INFO - Number of rows:  60837
[2025-03-11T21:13:33.626+0000] {logging_mixin.py:190} INFO - Number of columns:  28
[2025-03-11T21:15:00.204+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/airflow/dags/functions.py", line 118, in transform_data
    s3.put_object(Bucket= target_bucket_name, Key=object_key, Body=csv_data )
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/botocore/client.py", line 569, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/botocore/client.py", line 1023, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.NoSuchBucket: An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist
[2025-03-11T21:15:00.212+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-03-11T21:15:00.212+0000] {logging_mixin.py:190} INFO - Task start:2025-03-11 21:11:44.199953+00:00 end:2025-03-11 21:15:00.211975+00:00 duration:196.012022
[2025-03-11T21:15:00.212+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): tsk_transform_redfin_data> dag:<DAG: redfin_analytics_dag> dagrun:<DagRun redfin_analytics_dag @ 2025-03-11 21:02:19.139782+00:00: manual__2025-03-11T21:02:19.139782+00:00, state:running, queued_at: 2025-03-11 21:02:19.153009+00:00. externally triggered: True>
[2025-03-11T21:15:00.212+0000] {logging_mixin.py:190} INFO - Failure caused by An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist
[2025-03-11T21:15:00.213+0000] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=redfin_analytics_dag, task_id=tsk_transform_redfin_data, run_id=manual__2025-03-11T21:02:19.139782+00:00, execution_date=20250311T210219, start_date=20250311T211144, end_date=20250311T211500
[2025-03-11T21:15:00.240+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-11T21:15:00.240+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 108 for task tsk_transform_redfin_data (An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist; 17518)
Traceback (most recent call last):
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/airflow/dags/functions.py", line 118, in transform_data
    s3.put_object(Bucket= target_bucket_name, Key=object_key, Body=csv_data )
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/botocore/client.py", line 569, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/redfin_venv/lib/python3.12/site-packages/botocore/client.py", line 1023, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.NoSuchBucket: An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist
[2025-03-11T21:15:00.396+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-11T21:15:00.410+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-11T21:15:00.410+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
